tests/test_quantization.py

```diff
@@ -0,0 +1,696 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import unittest
+
+import torch
+import torch.nn as nn
+
+from accelerate import init_empty_weights
+from accelerate.test_utils import require_bnb, require_cuda, require_huggingface_suite, slow
+from accelerate.utils.bnb import get_quantized_model
+from accelerate.utils.dataclasses import BnbQuantizationConfig
+
+
+class BitsAndBytesConfigIntegration(unittest.TestCase):
+    def test_BnbQuantizationConfig(self):
+        with self.assertRaises(ValueError):
+            BnbQuantizationConfig(
+                load_in_8bit=True,
+                load_in_4bit=True,
+            )
+
+
+@slow
+@require_cuda
+@require_bnb
+@require_huggingface_suite
+class MixedInt8EmptyModelTest(unittest.TestCase):
+    # We keep the constants inside the init function and model loading inside setUp function
+
+    # We need to test on relatively large models (aka >1b parameters otherwise the quantiztion may not work as expected)
+    # Therefore here we use only bloom-1b3 to test our module
+    model_name = "marcsun13/bloom-1b7_with_lm_head"
+
+    # Constant values
+    EXPECTED_RELATIVE_DIFFERENCE = (
+        1.540025  # This was obtained on a Quadro RTX 8000 so the number might slightly change
+    )
```

Comment:
Put the comment above so that the auto-formatting doesn't split the line.

Reply:
fixed here [0dcc440](https://github.com/huggingface/accelerate/pull/1626/commits/0dcc440d902228a57e7148198758eba6b1b09f45)